client/src/pages/Map/Map.js

```diff
@@ -0,0 +1,189 @@
+import React, { useState, useEffect, useRef, useCallback } from "react";
+import ReactMapGL, { Marker, Popup } from "react-map-gl";
+import Geocoder from "react-map-gl-geocoder";
+import "react-map-gl-geocoder/dist/mapbox-gl-geocoder.css";
+import "mapbox-gl/dist/mapbox-gl.css";
+import { listSpots } from "../../api/spots";
+import Header from "../../components/Header";
+import Footer from "../../components/Footer";
+import mapMarkerYellow from "../../assets/icons/map-marker-yellow.svg";
+import mapMarkerRed from "../../assets/icons/map-marker-red.svg";
+import SpotEntryForm from "../../components/SpotEntryForm/SpotEntryForm";
+import MapMain from "../../components/MapMain/MapMain";
+import StyledMapContainer from "../../components/StyledMapContainer";
+import SpotEntryFormContainer from "../../components/SpotEntryFormContainer";
+import SpotPopupContainer from "../../components/SpotPopupContainer";
+
+const Map = () => {
+  const [spots, setSpots] = useState([]);
+  const [showPopup, setShowPopup] = useState({});
+  const [addSpot, setAddSpot] = useState(null);
+  const [viewport, setViewport] = useState({
+    height: "100%",
+    width: "100%",
+    latitude: 51.9565,
+    longitude: 10.2755,
+    zoom: 4.5,
+  });
+
+  // This code is used for the Geocoder
```

Comment:
A comment like this shouldn't be necessary if you make use of functions or custom hooks. 
See https://javascript.info/function-basics#functions-comments

Reply:
I checked our previous custom hooks (the one with the useLocalStorage) from the password manager, but do not really know how to do it here. I also checked the documentation on React. It would be cool to discuss it !